lib/mongoid/extensions/date.rb

```diff
@@ -50,20 +50,21 @@ def demongoize(object)
         #
         # @param [ Object ] object The object to mongoize.
         #
-        # @return [ Time ] The object mongoized.
+        # @return [ Time | nil ] The object mongoized or nil.
         def mongoize(object)
-          unless object.blank?
-            begin
-              if object.is_a?(String)
-                # https://jira.mongodb.org/browse/MONGOID-4460
-                time = ::Time.parse(object)
-              else
-                time = object.__mongoize_time__
-              end
+          return if object.blank?
+          begin
+            if object.is_a?(String)
+              # https://jira.mongodb.org/browse/MONGOID-4460
+              time = ::Time.parse(object)
+            else
+              time = object.__mongoize_time__
+            end
+            if time.acts_like?(:time)
               ::Time.utc(time.year, time.month, time.day)
-            rescue ArgumentError
-              nil
             end
+          rescue ArgumentError
```

Comment:
I would like this rescue to be only wrapped around the actual parse operations because it is very generic and could mask other problems.

Reply:
done